lib/features/mailbox_dashboard/data/datasource_impl/session_storage_composer_datasoure_impl.dart

```diff
@@ -1,47 +1,103 @@
 import 'dart:convert';
 import 'package:collection/collection.dart';
 import 'package:core/core.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/user_name.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
 import 'package:jmap_dart_client/jmap/mail/email/email.dart';
 import 'package:model/model.dart';
-import 'package:tmail_ui_user/features/mailbox_dashboard/data/model/composer_cache.dart';
+import 'package:tmail_ui_user/features/caching/utils/cache_utils.dart';
 import 'package:tmail_ui_user/features/mailbox_dashboard/data/datasource/session_storage_composer_datasource.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/data/model/composer_cache.dart';
+import 'package:tmail_ui_user/main/exceptions/exception_thrower.dart';
 import 'package:universal_html/html.dart' as html;
 
 class SessionStorageComposerDatasourceImpl
     extends SessionStorageComposerDatasource {
+  SessionStorageComposerDatasourceImpl(this._htmlTransform, this._exceptionThrower);
+
+  final HtmlTransform _htmlTransform;
+  final ExceptionThrower _exceptionThrower;
+
   @override
-  ComposerCache getComposerCacheOnWeb() {
-    try {
-      final result = html.window.sessionStorage.entries.firstWhereOrNull((e) => e.key == EmailActionType.reopenComposerBrowser.name);
-      if (result != null) {
-        final emailCache = ComposerCache.fromJson(jsonDecode(result.value));
-        return emailCache;
-      } else {
-        throw NotFoundInWebSessionException();
+  Future<ComposerCache> getComposerCacheOnWeb(
+    AccountId accountId,
+    UserName userName
+  ) async {
+    return Future.sync(() async {
+      try {
+        final keyWithIdentity = TupleKey(
+          EmailActionType.reopenComposerBrowser.name,
+          accountId.asString,
+          userName.value).toString();
+          
+        final result = html.window.sessionStorage.entries.firstWhereOrNull(
+          (entry) => entry.key == keyWithIdentity);
+        if (result != null) {
+          final composerCache = ComposerCache.fromJson(jsonDecode(result.value));
+          
+          return ComposerCache(
+            email: composerCache.email,
+            identity: composerCache.identity);
+        } else {
+          throw NotFoundInWebSessionException();
+        }
+      } catch (e) {
+        throw NotFoundInWebSessionException(errorMessage: e.toString());
       }
```

Comment:
IMO, we don't need `try/catch` here because there is already `catchError` outside

Reply:
Done